### DISCUSSION TITLE
Sample Python Script templates for newbies.

### DISCUSSION THREAD

**Discussion by jahagirdar** [freeplane/freeplane#952] [General]
The scripting document indicates that we can use python to write scripts that will execute within the freeplan environment.
But there is not documentation detailing how this can be achieved. A small example /quickstart guide will help a lot.

**Comment by macmarrum** [comment]
It's actually Jython that can be used for scripting.

There's some old documentation about Jython, which looks alright to me
https://docs.freeplane.org/attic/old-mediawiki-content/Scripting!_Other_languages.html

I'll just add that the `/lib/` directory is located in User directory. To open it, in Freeplane you can go to `Tools->Open user directory`.

If still in doubt, feel free to come back and ask.

**Comment by jahagirdar** [comment]
Thanks, I will try it out. and update here.

My goal:

I have put together a script in python which replicates docear's feature of pulling in highlights and comments from a pdf file. 
https://github.com/jahagirdar/litman/blob/master/litman/mm.py

This uses @nnako 's freeplane-python-io.

While the solution works, it requires reloading the mindmap in freemind. So the next step for me is to rewrite the script so that it can be called from within freemind on the loaded map

**Reply by macmarrum** [reply]
It's a great idea.

I am curious whether you'll be able to get around the challenges in running freeplane-python-io in Jython:
* freeplane-python-io is Python 3 project whereas Jython is Python 2
* freeplane-python-io depends on lxml, which itself depends on CPython

**Reply by jahagirdar** [reply]
Since Jython is Python2, I gave up on that approach. The base library for parsing pdf also has a javascript wrapper so someday I might rewrite it as a native JS plugin for Freeplane.
